### ISSUE TITLE
🛑 The Status of Periphery

### ISSUE THREAD

**Issue by ileitch** [peripheryapp/periphery#66]
Hey all, first off I'd like to apologize for not being responsive. To be fair to you all, it's time that I explain the status of Periphery and why I haven't been actively involved.

In Xcode 10.2 Apple introduced some large changes to source indexing, which unfortunately also introduced some bugs in the index output from SourceKit. I reported these issues to Apple and attempted to workaround them as best I could.

In Xcode 11 there were more indexing changes that again changed the format of the index output from SourceKit. It became very difficult to workaround these issues and get Periphery's test suite passing again. Frustrated with the situation, I reached out to some Apple engineers on twitter: https://twitter.com/peripheryappcom/status/1196059657183354880

The tl;dr is that Apple no longer supports the indexing output from SourceKit, since Xcode itself no longer uses it. They're ensuring that existing tests don't break, however I can only imagine their test coverage is lacking in many areas.

The recommended approach is to use https://github.com/apple/indexstore-db. Unfortunately this would require a near complete re-write of Periphery. There's also no guarantee that all of Periphery's analysis techniques can be adapted to IndexStoreDB. I personally have no desire to undertake this work, though I'm more than happy to provide support & guidance for anyone that wishes to attempt this.

So there you have it. Unless someone steps up to continue developing Periphery, I'm afraid it's pretty much dead.

**Comment by kateinoigakukun** [comment]
I agree that using indexstore-db instead is a better approach. I'll work on this because index-while-building doesn't require re-building projects with WMO and it enables to use of incrementally built products.
